ferrischat_webserver/src/users/delete_user.rs

```diff
@@ -1,6 +1,32 @@
 use actix_web::{web::Path, HttpResponse, Responder};
+use ferrischat_common::types::{InternalServerErrorJson, NotFoundJson};
 
 /// DELETE /api/v0/users/{user_id}
-pub async fn delete_user(Path(user_id): Path<i64>, _: crate::Authorization) -> impl Responder {
-    HttpResponse::NoContent()
+pub async fn delete_user(req: HttpRequest, auth: crate::Authorization) -> impl Responder {
+    let user_id = get_item_id!(req, "user_id");
+
+    if user_id != auth.0 {
+        return HttpResponse::Forbidden().finish();
+    }
+    let bigint_user_id = u128_to_bigdecimal!(user_id);
+
+    let db = get_db_or_fail!();
+    let resp = sqlx::query!(
+        "DELETE FROM users WHERE id = $1 RETURNING (id)",
+        bigint_user_id
+    )
+    .fetch_optional(db)
+    .await?;
```

Comment:
the ? here converts this into Ok(T) from `Result<T, E>` and returns `Err(E)` from the function, so also remove this one
```suggestion
    .await;
```